launch/launch/actions/execute_process.py

```diff
@@ -429,13 +442,17 @@ def __expand_substitutions(self, context):
         cwd = None
         if self.__cwd is not None:
             cwd = ''.join([context.perform_substitution(x) for x in self.__cwd])
-        env = None
+        env = {}
         if self.__env is not None:
-            env = {}
             for key, value in self.__env:
                 env[''.join([context.perform_substitution(x) for x in key])] = \
                     ''.join([context.perform_substitution(x) for x in value])
-
+        else:
+            env = dict(os.environ)
```

Comment:
The problem I have with the current logic is that `None` is never passed to the subprocess call, under any condition (as far as I can tell).

I think that if `self.__env` is `None` and `self.__additional_env` is also `None`, then `env` in this scope should be `None` as well.

Reply:
It will be effectively the same thing, but I'd really like a way to actually defer to the behavior of subprocess when env is None, which is currently not possible.